django/db/backends/mysql/operations.py

```diff
@@ -175,21 +175,54 @@ def random_function_sql(self):
 
     def sql_flush(self, style, tables, sequences, allow_cascade=False):
         # NB: The generated SQL below is specific to MySQL
-        # 'TRUNCATE x;', 'TRUNCATE y;', 'TRUNCATE z;'... style SQL statements
+        # 'TRUNCATE x;', 'TRUNCATE y;', 'DELETE FROM z;'... style SQL statements
         # to clear all tables of all data
+        # 'DELETE FROM' is used for small tables, 'TRUNCATE' is used for large tables
+
         if tables:
+            schema_name = self.connection.settings_dict['NAME']
+
+            with self.connection.cursor() as cursor:
+                cursor.execute("""
+                SELECT table_name, table_rows
+                FROM information_schema.tables
+                WHERE table_schema = %s AND table_name IN %s
```

Comment:
In MySQL introspection we use `table_schema =  DATABASE()`, I think we should use it here.